src/clusterfuzz/_internal/google_cloud_utils/storage.py

```diff
@@ -1351,12 +1352,17 @@ def _sign_urls_for_existing_file(
   return (download_url, delete_url)
 
 
+def _mappable_sign_urls_for_existing_file(url_and_include_delete_urls):
+  url, include_delete_urls = url_and_include_delete_urls
+  return _sign_urls_for_existing_file(url, include_delete_urls)
+
+
 def sign_urls_for_existing_files(urls,
                                  include_delete_urls) -> List[Tuple[str, str]]:
   logs.info('Signing URLs for existing files.')
-  result = [
-      _sign_urls_for_existing_file(url, include_delete_urls) for url in urls
-  ]
+  args = ((url, include_delete_urls) for url in urls)
```

Comment:
Is `include_delete_urls` the same value for the entire call? 

It's not clear to me why we need this `_mappable_sign_urls_for_existing_file` in that case, and all these list comprehensions. 

Can we avoid all of these and just do:

```python
pool.map(lambda url: _sign_urls_for_existing_file(url, include_delete_urls), urls)
```

?

Reply:
I can try, but the reason I'm programming like closures don't exist is because I think they've caused me issues with multiprocessing and we don't have test coverage of multiprocessing.

Reply:
Can you expand on the kind of issues you encountered?

It's not clear to me how using a lambda here would introduce multiprocessing issues. Was it to do with variable capture? 

In any case, having lots of these functions like `_mappable_sign_urls_for_existing_file` that only serve the purpose of changing the structure of some internal representation to make our implementation details fit seems non-ideal. We should either avoid this (e.g. via a lambda), or change our underlying data structures in a way that avoids this need.